File: docs/current_docs/snippets/documentation/go/object.go
package main

// The struct represents a single user of the system.
type MyModule struct {
	Name string
	Age  int
}

func New(
	// The name of the user.
	name string,
	// The age of the user.
	age int,
) *MyModule {
	return &MyModule{
		Name: name,
		Age:  age,
	}
}

File: docs/current_docs/snippets/documentation/typescript/object.ts
import { object } from "@dagger.io/dagger"

/**
 * The object represents a single user of the system.
 */
@object()
class MyModule {
  @func()
  name: string
  @func()
  age: number

  constructor(
    /**
     * The name of the user.
     */
    name: string,
    /**
     * The age of the user.
     */
    age: number,
  ) {
    this.name = name
    this.age = age
  }
}
